File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        int l,r,i,water=0;
        l=0,r=h.size()-1;

        while(l<r)
        {
            int area= min(h[l],h[r])*(r-l);

            water = max(water,area);

            if(h[l]<h[r])l++;
            else r--;
        }
        return water;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] a) {

        int i,j,n,maxi=0;
        n=a.length;
        
        i=0;j=0;
        while(j<n)
        {
           if(a[i]==a[j])
           {
            j++;
           } 
           else if (a[i]>a[j])
           {
            i++;
           }
           else
           {
            maxi=Math.max(maxi,a[j]-a[i]);
            j++;
           }
        }
        return maxi;
    }
}
